monitoring/rpc_stats_interceptor.go

```diff
@@ -0,0 +1,95 @@
+package monitoring
+
+import (
+	"expvar"
+	"fmt"
+	"time"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"github.com/google/trillian/util"
+)
+
+const (
+	nanosToMillisDivisor int64 = 1000000
+
+	requestCountMapName string = "requests-by-handler"
+	successCountMapName string = "success-by-handler"
+	errorCountMapName string = "errors-by-handler"
+	successLatencyTotalMapName string = "successful-total-latency-by-handler-ms"
+	errorLatencyTotalMapName string = "errored-total-latency-by-handler-ms"
+)
+
+type RpcStatsInterceptor struct {
+	baseName                           string
+	timeSource                         util.TimeSource
+	handlerRequestCountMap             *expvar.Map
+	handlerRequestSucceededMap         *expvar.Map
+	handlerRequestErrorsMap            *expvar.Map
+	handlerRequestSuccessfulLatencyMap *expvar.Map
+	handlerRequestFailedLatencyMap     *expvar.Map
+}
+
+func NewRpcStatsInterceptor(timeSource util.TimeSource, application, component string) *RpcStatsInterceptor {
+	return &RpcStatsInterceptor{baseName: fmt.Sprintf("%s/%s", application, component), timeSource: timeSource,
+		handlerRequestCountMap: new(expvar.Map).Init(),
+		handlerRequestSucceededMap: new(expvar.Map).Init(),
+		handlerRequestErrorsMap: new(expvar.Map).Init(),
+		handlerRequestSuccessfulLatencyMap: new(expvar.Map).Init(),
+		handlerRequestFailedLatencyMap: new(expvar.Map).Init()}
+}
+
+func (r RpcStatsInterceptor) nameForMap(name string) string {
+	return fmt.Sprintf("%s/%s", r.baseName, name)
+}
+
+// Publish must be called for stats to be visible. The expvar framework will prevent
+// multiple calls to Publish from succeeding.
+func (r RpcStatsInterceptor) Publish() {
+	expvar.Publish(r.nameForMap(requestCountMapName), r.handlerRequestCountMap)
+	expvar.Publish(r.nameForMap(successCountMapName), r.handlerRequestSucceededMap)
```

Comment:
Maybe make the consts which contain the exported names for the maps match the map names?
e.g.
```go
...r.nameForMap(requestSucceededMapName), r.handlerRequestSucceededMap)
```